blockchain/utxocache.go

```diff
@@ -640,7 +632,7 @@ func (s *utxoCache) Commit(view *UtxoViewpoint, node *blockNode) error {
 		if node != nil && isBIP0030Node(node) {
 			// Use the entry from the view since we want to overwrite the
 			// cached entry.
-			cachedEntry = entry.Clone()
```

Comment:
The cloning also showed up a lot in my memory profiles, what makes it no longer required? Fact that these never get mutated anyway?

Reply:
Ah yeah that was my finding. If the entry is used internally within the blockchain package, we can make sure it's not modified so no need to Clone(). Also, I wasn't sure how much the Clone() does to help with immutability since the pkscript is still mutable as the Clone() here is a shallow clone.

I saw that the mining code does fetch for entries from the tx and one of the tests fails if I don't call Clone() for the entries that are fetched by public functions. But not calling Clone() for functions that are private didn't have any effect and all the code that calls private fetch functions do not mutate the entries.